eogrow/pipelines/zipmap.py

```diff
@@ -0,0 +1,133 @@
+    folder_key: str = Field("The storage manager key pointing to the folder from which to load data.")
+    include_bbox_and_timestamp = Field(
+        True,
+        description="Auto loads BBOX and (if the features is temporal) TIMESTAMP.",
+    )
+
+
+class ZipMapPipeline(Pipeline):
+    class Schema(Pipeline.Schema):
+        input_features: List[InputFeatureSchema] = Field(
+            description="The specification for all the features to be loaded."
+        )
+
+        zipmap_import_path: str = Field(
+            description="Import path of the callable with which to process the loaded features."
+        )
+        params_model: Optional[str] = Field(
+            description=(
+                "Optional import path for the pydantic model class, with which to parse and validate the parameters for"
+                " the callable. The model will be used to parse the params and then unpacked back into a dictionary, "
+                " which is passed to the callable as `**params`."
+            )
+        )
+        params: Dict[str, Any] = Field(
+            default_factory=dict, description="Any keyword arguments to be passed to the zipmap function."
+        )
+
+        @validator("params")
+        def parse_params(cls, v, values):  # type: ignore[no-untyped-def]
+            """Parse the parameters according to model, but returning as a dictionary to allow `**kwargs` passing."""
+            if values.get("params_model"):
+                params_model: BaseSchema = import_object(values["params_model"])
+                return params_model.parse_obj(v).dict()
+            return v
+
+        output_folder_key: str = Field(
+            description="The storage manager key pointing to the output folder for the algorithm pipeline."
+        )
+        output_feature: FeatureSpec
+
+        compress_level: int = Field(1, description="Level of compression used in saving eopatches.")
+
+    config: Schema
+
+    def filter_patch_list(self, patch_list: List[str]) -> List[str]:
+        """EOPatches are filtered according to existence of new features"""
+        # Note: does not catch missing BBox or Timestamp
+        filtered_patch_list = get_patches_with_missing_features(
+            self.storage.filesystem,
+            self.storage.get_folder(self.config.output_folder_key),
+            patch_list,
+            [self.config.output_feature],
+        )
+
+        return filtered_patch_list
+
+    def get_load_nodes(self) -> List[EONode]:
+        """Prepare all nodes with load tasks."""
+        load_schema: DefaultDict[str, Set[FeatureSpec]] = defaultdict(set)
+        for input_feature in self.config.input_features:
+            features_to_load = load_schema[input_feature.folder_key]
+            features_to_load.add(input_feature.feature)
+
+            if input_feature.include_bbox_and_timestamp:
+                features_to_load.add(FeatureType.BBOX)
```

Comment:
What is the main reason that we are handling the bbox and timestamps separately? 


 It does add quite some complexity to the code so I want to understand the motivations behind it. Is it just to not load data that we don't need? Or is there another reason?

Reply:
Unlike `SaveTask`, loading demands the features exist.
So ofcourse if the features are not temporal, it makes no sense to load the timestamps (a lot less IO calls). But in some cases when we have storages with lots of subfolders, some don't contain a BBox, and this was meant to avoid such things.

We again weren't sure whether this should be always on, or just on by default.

Reply:
I understand. I think you are correct, let's leave it like it is.